pvlib/iotools/pvgis.py

```diff
@@ -23,6 +23,348 @@
+    if outputformat == 'json':
+        src = res.json()
+        return _parse_pvgis_hourly_json(src, map_variables=map_variables)
+    elif outputformat == 'csv':
+        with io.StringIO(res.content.decode('utf-8')) as src:
+            return _parse_pvgis_hourly_csv(src, map_variables=map_variables)
+    else:
+        # this line is never reached because if outputformat is not valid then
+        # the response is HTTP/1.1 400 BAD REQUEST which is handled earlier
+        pass
+    return data
+
+
+def _parse_pvgis_hourly_json(src, map_variables):
+    inputs = src['inputs']
+    metadata = src['meta']
+    data = pd.DataFrame(src['outputs']['hourly'])
+    data.index = pd.to_datetime(data['time'], format='%Y%m%d:%H%M', utc=True)
+    data = data.drop('time', axis=1)
+    data = data.astype(dtype={'Int': 'int'})  # The 'Int' column to be integer
+    if map_variables:
+        data.rename(columns=PVGIS_VARIABLE_MAP, inplace=True)
+    return data, inputs, metadata
+
+
+def _parse_pvgis_hourly_csv(src, map_variables):
+    # The first 4 rows are latitude, longitude, elevation, radiation database
+    inputs = {}
+    # 'Latitude (decimal degrees): 45.000\r\n'
+    inputs['latitude'] = float(src.readline().split(':')[1])
+    # 'Longitude (decimal degrees): 8.000\r\n'
+    inputs['longitude'] = float(src.readline().split(':')[1])
+    # Elevation (m): 1389.0\r\n
+    inputs['elevation'] = float(src.readline().split(':')[1])
+    # 'Radiation database: \tPVGIS-SARAH\r\n'
+    inputs['radiation_database'] = str(src.readline().split(':')[1]
+                                       .replace('\t', '').replace('\n', ''))
+    # Parse through the remaining metadata section (the number of lines for
+    # this section depends on the requested parameters)
+    while True:
+        line = src.readline()
+        if line.startswith('time,'):  # The data header starts with 'time,'
+            # The last line of the metadata section contains the column names
+            names = line.strip().split(',')
+            break
+        # Only retrieve metadata from non-empty lines
+        elif (line != '\n') & (line != '\r\n'):
+            inputs[line.split(':')[0]] = str(line.split(':')[1]
+                                             .replace('\n', '')
+                                             .replace('\r', '').strip())
+    # Save the entries from the data section to a list, until an empty line is
+    # reached an empty line. The length of the section depends on the request
+    data_lines = []
+    while True:
+        line = src.readline()
+        if (line == '\n') | (line == '\r\n'):
+            break
+        else:
+            data_lines.append(line.strip().split(','))
+    data = pd.DataFrame(data_lines, columns=names)
+    data.index = pd.to_datetime(data['time'], format='%Y%m%d:%H%M', utc=True)
+    data = data.drop('time', axis=1)
+    if map_variables:
+        data.rename(columns=PVGIS_VARIABLE_MAP, inplace=True)
```

Comment:
I try to avoid `inplace=True` because pandas is so inconsistent about it. I think Wes McKinney once said it was a bad idea. In any case, it's more characters than `data = `!

Reply:
Thank you for the explanation, I just keep learning! :)

Consider it changed.